### PR TITLE
OSD sandbox test case

### PR DESCRIPTION
**Pull request by tarukumar**
Signed-off-by: Tarun Kumar <takumar@redhat.com>

### CODE REVIEW

**Comment by tarukumar** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/621/  all TCs passed

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Resource        ../../Resources/Common.robot
+Resource        ../../Resources/Page/ODH/ODHDashboard/ODHDashboard.resource
+Library         String
+Library         SeleniumLibrary
+Suite Setup     Begin Web Test
+Suite Teardown  End Web Test
+
+
+*** Variables ***
+${SANDBOX_AUTH_TYPE}   *******
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:12]: This variables probably should be in test-variables.yaml

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+${SANDBOX_USERNAME}    *******
+${SANDBOX_PASSWORD}    *******
+${SANDBOX_OCP_URL}     *******
+
+
+*** Test Cases ***
+Verify Application Switcher Have Only Opneshift Console Link
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:19]: Typo in OpenShift

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    [Documentation]    Application switcher should have only
+    ...    openshift conosle link
+    [Tags]     ODS-309
+    ...        Sandbox
+    Check Openshift Console
+
+Verify ISV integration enablement IS Disabled
+    [Documentation]    All the isv integration
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:27]: Verifies that all the ISV integration components are disabled in the Dashboard

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    ...        Sandbox
+    Click Link    Explore
+    Capture Page Screenshot   explor.png
+    Wait Until Cards Are Loaded
+    Sleep    5s
+    ${link_elements}  Get WebElements    //article[@class="pf-c-card odh-card m-disabled"]
+    ${length}      Get Length    ${link_elements}
+    Run Keyword If  ${length} != ${11}    Fail     '${length}' tiles in Explore section is disabled
+
+Verify Support Link Doesn't Exist
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:39]: Maybe better `Verify All Cards Have "Support" Link Hidden`

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    [Documentation]   Only documentaion link is present
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:40]: Verify that all cards have the support link disabled/hidden, as in sandbox they should have only the documentation link

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    ...       Sandbox
+    Click Element    xpath=//*[@id="toggle-id"]
+    ${links_avialble}   Get WebElements
+    ...    //a[@class="odh-dashboard__external-link pf-c-dropdown__menu-item" and not(starts-with(@href, '#'))]
+    FOR  ${link}  IN  @{links_avialble}
+         ${href}    Get Element Attribute    ${link}    href
+         Run Keyword And Continue On Failure   Should Not Contain   ${href}   support   ignore_case=True
+    END
+
+Verify only small size is available for sandbox enviroinment
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:51]: TitleCase missing. Typo in environment

Maybe better `Verify JupyterHub Spawner Only Allows Small Server Size In Sandbox Environment`

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    [Documentation]    Only Default and Small size is present
+    ...    on spawner
+    [Tags]     ODS-528
+    ...        Sandbox
+    Navigate To Page    Applications    Enabled
+    Launch JupyterHub Spawner From Dashboard
+    Match Small And Default Container Size
+
+Verify that idle Jupyterhub servers are culled
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:60]: TitleCase missing. Sandbox tag missing

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    ${jl_title}     Get Title
+    Switch Window    title=Red Hat OpenShift Data Science Dashboard
+    Sleep    24h
+    Switch Window     title=${jl_title}
+    Wait Until Keyword Succeeds    120    2
+    ...    Page Should Contain    Server unavailable or unreachable
+
+
+*** Keywords ***
+Get RHODS URL From OCP
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:74]: I think this should be in ODS.robot, and just use the OpenShiftLibrary to get the route (or make the url yourself from OCP_CONSOLE_URL

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+     ${length}         Get Length    ${link_elements}
+     ${href}   Get Element Attribute    ${link_elements}    href
+     ${length}         Get Length    ${link_elements}
+     IF    ${length} == ${1}
+           Should Contain      ${href}    console-openshift   ignore_case=True
+     ELSE
+           Run Keyword And Continue On Failure    Fail    More than one link is present
+     END
+
+Match Small And Default Container Size
```
Comment (jgarciao) [inline tests/Tests/700__sandbox/700__sandbox.robot:99]: I'm not 100% sure but in JupyterHubSpawner you migh find a keyword that does this or similar. 

If not, you could add there a keyword  `Available Container Sizes Should Be` or `Verify Available Container Sizes` keyword, passing a list of allowed sizes as parameter

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 7 above)
[the remark this answers is not in the document]
Comment (jgarciao) [reply tests/Tests/700__sandbox/700__sandbox.robot:60]: You can add tag `Execution-Time-Over-1d`

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 4 above)
Comment (bdattoma) [reply tests/Tests/700__sandbox/700__sandbox.robot:39]: I think this is not about the cards, but it refers to the "support link" of the Dashboard

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    ${link_elements}  Get WebElements    //article[@class="pf-c-card odh-card m-disabled"]
+    ${length}      Get Length    ${link_elements}
+    Run Keyword If  ${length} != ${11}    Fail     '${length}' tiles in Explore section is disabled
+
+Verify Support Link Doesn't Exist
+    [Documentation]   Only documentaion link is present
+    [Tags]    ODS-526
+    ...       Sandbox
+    Click Element    xpath=//*[@id="toggle-id"]
+    ${links_avialble}   Get WebElements
```
Comment (bdattoma) [inline tests/Tests/700__sandbox/700__sandbox.robot:44]: typo in the var name

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    ...    //a[@class="odh-dashboard__external-link pf-c-dropdown__menu-item" and not(starts-with(@href, '#'))]
+    FOR  ${link}  IN  @{links_avialble}
+         ${href}    Get Element Attribute    ${link}    href
+         Run Keyword And Continue On Failure   Should Not Contain   ${href}   support   ignore_case=True
+    END
```
Comment (bdattoma) [inline tests/Tests/700__sandbox/700__sandbox.robot:49]: could you move the for loop out of the test case?
Based on the doc tag, maybe you should add the check that the documentation link is present

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+    Match Small And Default Container Size
+
+Verify that idle Jupyterhub servers are culled
+    [Documentation]   Jupyterhub should be culled
+    ...    after 24 hours
+    [Tags]     ODS-547
+    Spawn Notebook With Arguments
+    ${jl_title}     Get Title
+    Switch Window    title=Red Hat OpenShift Data Science Dashboard
+    Sleep    24h
```
Comment (bdattoma) [inline tests/Tests/700__sandbox/700__sandbox.robot:67]: shouldn't we change the timeout to a lower one? I would assume that if the timeout should work either cases

File: tests/Tests/700__sandbox/700__sandbox.robot [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+    ...     OCP console
+    Switch To Administrator Perspective
+    Click Element     //button[@aria-label="Application launcher"]
+    Wait Until Element Is Visible    //a[@data-test="application-launcher-item"]
+    ${link_elements}  Get WebElements
+    ...     //a[@data-test="application-launcher-item" and starts-with(@href,'https://rhods')]
+    ${href}  Get Element Attribute    ${link_elements}    href
+    [Return]   ${href}
+
+Check Openshift Console
```
Comment (bdattoma) [inline tests/Tests/700__sandbox/700__sandbox.robot:85]: maybe you could find a more self-explanatory title for this kw

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 1 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Removed

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 2 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 6 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 7 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: ack

[the remark this answers is not in the document]
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 8 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 9 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 10 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Done

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 11 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: i have move it to keyword

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 13 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot]: Change it

File: tests/Tests/700__sandbox/700__sandbox.robot (hunk 12 above)
Comment (tarukumar) [reply tests/Tests/700__sandbox/700__sandbox.robot:56]: for now i always verify after 24hr we could do that later however once culler is there we can modify it for now i will try to keep like this

**Review by jgarciao — APPROVED** [review]
lgtm

**Review by bdattoma — APPROVED** [review]
lgtm

**Pull request merged**